### ISSUE TITLE
Java.lang.OUtOfMEmoryError: Failed To allocate a ... byte allocation with ... bytes and 70mb until OOM

### ISSUE THREAD

**Issue by amrsunja** [rafid08/media_picker_widget#2]
In my xiaomi phone I was received this failure when I try to get images and videos from gallery

**Comment by XuanTung95** [comment]
Currently, this library read bytes of all media files, causing OOM when the file is too big. Should only load the thumbnail to display.

```
Future<Media> convertToMedia({required AssetEntity media}) async {
  Media convertedMedia = Media();
  convertedMedia.file = await media.file;
  convertedMedia.mediaByte = await media.originBytes; // <-- here
  convertedMedia.thumbnail =
      await media.thumbnailDataWithSize(ThumbnailSize(200, 200));
  convertedMedia.id = media.id;
  convertedMedia.size = media.size;
  convertedMedia.title = media.title;
  convertedMedia.creationTime = media.createDateTime;

  MediaType mediaType = MediaType.all;
  if (media.type == AssetType.video) mediaType = MediaType.video;
  if (media.type == AssetType.image) mediaType = MediaType.image;
  convertedMedia.mediaType = mediaType;

  return convertedMedia;
}
```

**Comment by XuanTung95** [comment]
Hello @rafid08
I think your package is awesome, but I noticed some problems: 
- You read mediaByte for every image/video, which is not necessary and cause OOM
- Almost the data of Media come from AssetEntity. Why don't you return AssetEntity instead? AssetEntity contains more info and has useful methods for user.
- Some fetch data method is in build() method, which is not safe because build() can be called multiple time by the framework:
"Build method can potentially be called in every frame and should not have any side effects beyond building a widget."

**Comment by rafidgotit** [comment]
Hey @XuanTung95 
Thank you so much for your valuable suggestions.
Also I'm extremely sorry for the issue you're facing.
The thing is, the time when I developed this package, I was not that expert in flutter. Now I really want to refactor the whole project but I'm currently so busy with some of my other projects. I'm not getting that much time to refactor the project.
I'd be glad if you would like do some contributions to this project...

**Comment by XuanTung95** [comment]
Hi @rafid08,
I would love to contribute to this project.
Let me see what I can do.

**Comment by reliance-engineer** [comment]
Hi, I'm having this same issue, will be a solution for this soon??

**Comment by rafidgotit** [comment]
This issue is solved in the last update of `1.0.1`
I'm extremely sorry for being late solving this major issue.
